src/parsers/xyz.ts

```diff
@@ -36,59 +37,79 @@ function validateLine(xyzLine, index) {
 
 /**
  * Validates that passed string is well-formed XYZ file.
- * @param xyzTxt {String}
  */
-export function validate(xyzTxt) {
+export function validate(xyzTxt: string) {
+    // The @types/underscore.string package does not provide interfaces for function chaining at the moment. Disable TS here
+    // @ts-ignore
     s(xyzTxt)
         .trim()
         .lines()
-        .filter((x) => x.trim() !== "")
+        .filter((x: string) => x.trim() !== "")
         .forEach(validateLine);
 }
 
+export interface ParsedObject {
+    element: string;
+    coordinates: Vector;
+    constraints: [boolean, boolean, boolean];
+}
+
 /**
  * Parses XYZ line and returns an object.
- * @param {String} line - line of text
- * @return {Object}
+ * @param line - line of text
  */
-function _parseXYZLineAsWords(line) {
+function _parseXYZLineAsWords(line: string): ParsedObject {
     const words = s.words(line);
+    const constraint = (n: number) => parseInt(`${n}`, 10) !== 0;
     return {
         element: words[0],
         coordinates: [+words[1], +words[2], +words[3]],
         // Below maps zero values to false (atom is fixed) and non-zero values to true (atom is moving)
-        constraints: [+words[4], +words[5], +words[6]].map((e) => parseInt(e, 10) !== 0),
+        constraints: [constraint(+words[4]), constraint(+words[5]), constraint(+words[6])],
     };
 }
 
+export interface BasisConfig {
```

Comment:
Should be from ESSE

Reply:
For some reason BasisSchema from esse is different from return value of Basis.toJSON:
```ts
export interface BasisSchema {
  elements: AtomicElements[];
  coordinates: AtomicCoordinate[];
  name?: string;
  units?: string;
  bonds?: BondsSchema;
}
```
And
```ts
toJSON(): BasisJSON {
    const json: BasisJSON = {
        elements: this.elements,
        coordinates: this.coordinates,
        units: this.units,
        cell: this.cell,
    };

    return JSON.parse(JSON.stringify(json));
}
```

Reply:
@timurbazhirov to review

Reply:
Also `ConstrainedBasis.toJSON()` returns extended `BasisJSON` with `constraints` property 
```ts
export interface BasisJSON {
    elements: ObjectWithIdAndValue<string>[];
    coordinates: ObjectWithIdAndValue<Coordinate>[];
    units: string;
    cell: Vector[];
}

export interface ConstrainedBasisJSON extends BasisJSON {
    constraints: ObjectWithIdAndValue<ConstraintValue>[];
}
```